packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -723,40 +727,61 @@ class _RangeSliderState extends State<RangeSlider> with TickerProviderStateMixin
       mouseCursor: effectiveMouseCursor,
       child: CompositedTransformTarget(
         link: _layerLink,
-        child: _RangeSliderRenderObjectWidget(
-          values: _unlerpRangeValues(widget.values),
-          divisions: widget.divisions,
-          labels: widget.labels,
-          sliderTheme: sliderTheme,
-          textScaleFactor: effectiveTextScale,
-          screenSize: screenSize(),
-          onChanged: _enabled && (widget.max > widget.min) ? _handleChanged : null,
-          onChangeStart: widget.onChangeStart != null ? _handleDragStart : null,
-          onChangeEnd: widget.onChangeEnd != null ? _handleDragEnd : null,
-          state: this,
-          semanticFormatterCallback: widget.semanticFormatterCallback,
-          hovering: _hovering,
+        child: OverlayPortal(
+          controller: _valueIndicatorOverlayPortalController,
+          overlayChildBuilder: (BuildContext context) {
+            return _buildValueIndicator(sliderTheme.showValueIndicator);
+          },
+          child: _RangeSliderRenderObjectWidget(
+            values: _unlerpRangeValues(widget.values),
+            divisions: widget.divisions,
+            labels: widget.labels,
+            sliderTheme: sliderTheme,
+            textScaleFactor: effectiveTextScale,
+            screenSize: screenSize(),
+            onChanged: _enabled && (widget.max > widget.min) ? _handleChanged : null,
+            onChangeStart: widget.onChangeStart != null ? _handleDragStart : null,
+            onChangeEnd: widget.onChangeEnd != null ? _handleDragEnd : null,
+            state: this,
+            semanticFormatterCallback: widget.semanticFormatterCallback,
+            hovering: _hovering,
+          ),
         ),
       ),
     );
   }
 
   final LayerLink _layerLink = LayerLink();
-
-  OverlayEntry? overlayEntry;
+  Widget _buildValueIndicator(ShowValueIndicator? showValueIndicator) {
+    late final Widget valueIndicator = CompositedTransformFollower(
```

Comment:
```suggestion
    late final Widget valueIndicator = CompositedTransformFollower(
```

Is `late` required since the value is assigned here.

Reply:
If `valueIndicator ` is never accessed, the assignment expression for `valueIndicator` will not be executed. It is not necessarily required to create it in the subsequent code.

Reply:
the `late` isn't required to execute this code.